atmel-samd/tick.c

```diff
@@ -29,18 +44,21 @@ void timer_tick(const struct timer_task *const timer_task) {
 }
 
 void tick_init() {
-    #ifdef SAMD21
-    _pm_enable_bus_clock(PM_BUS_APBC, TC5);
-    #endif
-    // _gclk_enable_channel(TC5_GCLK_ID, GCLK_SOURCE_DFLL48M);
-
-    // timer_init(&ms_timer, TC5, _tc_get_timer());
+    SysTick->LOAD = ((CONF_CPU_FREQUENCY / 1000 - 1) << SysTick_LOAD_RELOAD_Pos);
```

Comment:
Could you add a TODO to replace this eventually, or have this call a HAL routine that returns the frequency?  Could call `common_hal_mcu_processor_get_frequency()`.